sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts

```diff
@@ -52,12 +161,358 @@ export class InstitutionInstitutionsController extends BaseController {
+   * @param payload
+   * @param user
+   * @returns Primary identifier of the created resource.
+   */
+  @ApiUnprocessableEntityResponse({
+    description:
+      "User to be added either not found in account service or does not belong to same institution of logged in user.",
+  })
+  @IsInstitutionAdmin()
+  @Post("user")
+  async createInstitutionUserWithAuth(
+    @Body() payload: InstitutionUserAPIInDTO,
+    @UserToken() user: IInstitutionUserToken,
+  ): Promise<PrimaryIdentifierAPIOutDTO> {
+    // Get institution
+    const institution = await this.institutionService.getInstituteByUserName(
+      user.userName,
+    );
+
+    // Find user on BCeID Web Service
+    const bceidUserAccount = await this.accountService.getAccountDetails(
+      payload.userId,
+    );
+    if (!bceidUserAccount) {
+      throw new UnprocessableEntityException(
+        "User to be added not found on BCeID Web Service.",
+      );
+    }
+    // Check if the user being added to th institution belongs to the institution.
+    if (
+      institution.guid.toLowerCase() !==
+      bceidUserAccount.institution.guid.toLowerCase()
+    ) {
+      throw new UnprocessableEntityException(
+        "User to be added not found under the institution.",
+      );
+    }
+
+    /** A legal signing authority role can be added to only one user per institution */
+    const addLegalSigningAuthorityExist = payload.permissions.some(
+      (role) => role.userRole === InstitutionUserRoles.legalSigningAuthority,
+    );
+
+    if (addLegalSigningAuthorityExist) {
+      const legalSigningAuthority =
+        await this.institutionService.checkLegalSigningAuthority(
+          institution.id,
+        );
+
+      if (legalSigningAuthority) {
+        throw new UnprocessableEntityException(
+          LEGAL_SIGNING_AUTHORITY_EXIST,
+          LEGAL_SIGNING_AUTHORITY_MSG,
+        );
+      }
+    }
+
+    // Create the user user and the related records.
+    const createdInstitutionUser =
+      await this.institutionService.createInstitutionUser(
+        institution.id,
+        bceidUserAccount,
+        payload,
+      );
+
+    return { id: createdInstitutionUser.id };
+  }
+
+  /**
+   * Get Institution user types and roles.
+   * @returns Institution user types and roles.
+   */
+  @IsInstitutionAdmin()
+  @Get("user-types-roles")
+  async getUserTypesAndRoles(): Promise<InstitutionUserTypeAndRoleAPIOutDTO> {
+    return this.institutionService.getUserTypesAndRoles();
+  }
+
+  /**
+   * Get institution user by user name(guid).
+   * @param userName
+   * @returns
+   */
+  @ApiNotFoundResponse({
+    description: "User not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description: "User not active.",
+  })
+  @IsInstitutionAdmin()
+  @Get("user/:userName")
+  async getInstitutionUserByUserName(
+    @Param("userName") userName: string,
+  ): Promise<InstitutionUserAPIOutDTO> {
+    // Get institutionUser
+    const institutionUser =
```

Comment:
It seems that there is an authorization check missing here. How can we ensure that the user has access to the _userName_ requested? If not part of the PR please add a TODO here.

Reply:
What is the missing authorization check ?

Reply:
I got the point after seeing the comment below.